onnxruntime/core/graph/schema_registry.cc

```diff
@@ -69,7 +70,8 @@ common::Status OnnxRuntimeOpSchemaRegistry::RegisterOpSchemaInternal(ONNX_NAMESP
             << op_schema.line()
             << ", but it is already registered from file "
             << schema.file() << " line " << schema.line() << std::endl;
-    return common::Status(common::ONNXRUNTIME, common::INVALID_ARGUMENT, ostream.str());
+    LOGS_DEFAULT(WARNING) << ostream.str();
+    return common::Status::OK();  // an op with the same name can be registered for multiple execution providers
```

Comment:
But the registration didn't happen even you returned OK? The kernel won't be used.

Reply:
i think just one opschema needs to be registered. so if it already exists, don't fail, which can happen for custom ops from multiple EPs.

Reply:
only issue is distinguishing multiple registration of the same opschema from same EP vs different EPs.

Reply:
Should we compare if the two schema equals or not? If they're not exactly the same, we should make it fail?

Reply:
@snnn the kernels get registered before the schemas. See [this](https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/core/session/custom_ops.cc#L140). 
@jywu-msft don't have the EP in this context, which is why i logged a warning.

Reply:
> Should we compare if the two schema equals or not? If they're not exactly the same, we should make it fail?

Possibly. The current code, however, is detecting duplicates based on this [map](https://github.com/microsoft/onnxruntime/blob/master/include/onnxruntime/core/graph/schema_registry.h#L15). The scope of the current PR does not intend to change this duplication detection strategy. It merely logs a warning when a duplicate is detected instead of throwing an error.